### DISCUSSION TITLE
Using within WSL(1)

### DISCUSSION THREAD

**Discussion by thestumbler** [stlink-org/stlink#1314] [Q&A (resolved)]
Not really a project, but a tip for using these STLink tools in WSL. 

I have to use Windows on a particular client's project. I have been using WSL(1) with reasonable success, but recently wanted to use these STLink tools (so some make files would be more portable between OSS). It seems that you can't easily (or at all?) use them in WSL. I was about to give up when I remembered something I rarely use -- you can run Windows .exe files within WSL. 

So I just installed the tool in Windows, and made symbolic links to them in WSL. Now everything works like a charm. 

Note: I don't know if/how this applies to WSL2. I started this project before WSL2 existed, and I was hesitant to switch because I reply heavily on accessing the serial ports. I haven't yet bothered to try WSL2 so have no experience one way or the other.

**Comment by Nightwalker-87** [comment]
One may test this in WSL2 as well.
However, my general personal experience with WSL is not the very best, regarding it's implementation within Windows.

**Comment by Nightwalker-87** [comment]
Closing this discussion as resolved.
